crates/turborepo-lib/src/turbo_json/future_flags.rs

```diff
@@ -0,0 +1,45 @@
+//! Future flags for enabling experimental or upcoming features
+//!
+//! This module contains the `FutureFlags` structure which allows users to
+//! opt-in to experimental features before they become the default behavior.
+//!
+//! ## Usage
+//!
+//! Future flags can be configured in the root `turbo.json`:
+//!
+//! ```json
+//! {
+//!   "futureFlags": {
+//!     "turboExtends": true
+//!   }
+//! }
+//! ```
+//!
+//! Note: Future flags are only allowed in the root `turbo.json` and will cause
+//! an error if specified in workspace packages.
+
+use biome_deserialize_macros::Deserializable;
+use serde::Serialize;
+use struct_iterable::Iterable;
+
+/// Future flags configuration for experimental features
+///
+/// Each flag represents an experimental feature that can be enabled
+/// before it becomes the default behavior in a future version.
+#[derive(Serialize, Default, Debug, Copy, Clone, Iterable, Deserializable, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct FutureFlags {
+    /// Enable `$TURBO_EXTENDS$`
+    ///
+    /// When enabled, allows using `$TURBO_EXTENDS$` in array fields.
+    /// This will change the default behavior of overriding the field to instead
+    /// append.
+    pub turbo_extends: bool,
```

Comment:
Non-blocking: Let's be slightly more descriptive on this flag name since we're about to have `$TURBO_EXTENDS$` and, separately, make the `extends` key more powerful. I could see a user being confused about which this refers to.